springwolf-plugins/springwolf-amqp-plugin/src/main/java/io/github/stavshamir/springwolf/producer/SpringwolfAmqpProducer.java

```diff
@@ -1,28 +1,58 @@
 package io.github.stavshamir.springwolf.producer;
 
+import com.asyncapi.v2.binding.amqp.AMQPChannelBinding;
+import com.asyncapi.v2.binding.amqp.AMQPOperationBinding;
+import com.asyncapi.v2.model.channel.ChannelItem;
+import com.asyncapi.v2.model.channel.operation.Operation;
+import io.github.stavshamir.springwolf.asyncapi.ChannelsService;
 import lombok.Getter;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.amqp.rabbit.core.RabbitTemplate;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
 
+import java.util.List;
 import java.util.Map;
 
 @Slf4j
 @Service
 public class SpringwolfAmqpProducer {
 
+    private final ChannelsService channelsService;
     private final RabbitTemplate rabbitTemplate;
 
     @Getter
     private boolean isEnabled = true;
 
-    public SpringwolfAmqpProducer(@Autowired RabbitTemplate rabbitTemplate) {
-        this.rabbitTemplate = rabbitTemplate;
+    public SpringwolfAmqpProducer(ChannelsService channelsService, List<RabbitTemplate> rabbitTemplates) {
+        this.channelsService = channelsService;
+        this.rabbitTemplate = rabbitTemplates.get(0);
     }
 
     public void send(String channelName, Map<String, Object> payload) {
-        rabbitTemplate.convertAndSend(channelName, payload);
+        ChannelItem channelItem = channelsService.getChannels().get(channelName);
+        String exchange = "";
+        if (channelItem.getBindings() != null && channelItem.getBindings().containsKey("amqp")) {
+            AMQPChannelBinding channelBinding = (AMQPChannelBinding) channelItem.getBindings().get("amqp");
+            if (channelBinding.getExchange() != null && channelBinding.getExchange().getName() != null) {
+                exchange = channelBinding.getExchange().getName();
+            }
+        }
+
+        String routingKey = "";
+        Operation operation = channelItem.getSubscribe() != null ? channelItem.getSubscribe() : channelItem.getPublish();
+        if (operation != null && operation.getBindings() != null && operation.getBindings().containsKey("amqp")) {
+            AMQPOperationBinding operationBinding = (AMQPOperationBinding) operation.getBindings().get("amqp");
+            if (!CollectionUtils.isEmpty(operationBinding.getCc())) {
+                routingKey = operationBinding.getCc().get(0);
+            }
+        }
```

Comment:
These ifs and null checks are not needed if bindings are constructed from annotations, but in the case when ProducerData is created manually, they are needed. So when the annotation for the Producer is added and ProducerData is deprecated and removed, the null checks can also be removed.